django/db/backends/sqlite3/operations.py

```diff
@@ -35,17 +35,18 @@ def check_expression_support(self, expression):
         bad_fields = (fields.DateField, fields.DateTimeField, fields.TimeField)
         bad_aggregates = (aggregates.Sum, aggregates.Avg, aggregates.Variance, aggregates.StdDev)
         if isinstance(expression, bad_aggregates):
-            try:
-                output_field = expression.input_field.output_field
-                if isinstance(output_field, bad_fields):
-                    raise NotImplementedError(
-                        'You cannot use Sum, Avg, StdDev and Variance aggregations '
-                        'on date/time fields in sqlite3 '
-                        'since date/time is saved as text.')
-            except FieldError:
-                # not every sub-expression has an output_field which is fine to
-                # ignore
-                pass
+            for expr in expression.get_source_expressions():
+                try:
+                    output_field = expr.output_field
+                    if isinstance(output_field, bad_fields):
+                        raise NotImplementedError(
+                            'You cannot use Sum, Avg, StdDev and Variance aggregations '
+                            'on date/time fields in sqlite3 '
+                            'since date/time is saved as text.')
```

Comment:
I do think keeping the try block as small as possible is helpful for understanding how the code is supposed to work.